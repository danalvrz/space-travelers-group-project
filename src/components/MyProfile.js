import React from 'react';
import { useSelector } from 'react-redux';

function MyProfile() {
  const missions = useSelector((state) => state.missions);
  const myMissionsArr = [];

  const MyMissions = (myMissionsData) => {
    if (myMissionsData.missions !== undefined) {
      myMissionsData.missions.filter((mission) => mission.member === true).forEach((mission) => {
        myMissionsArr.push(
          <tr key={mission.mission_id} className="border h-14">
            <td className="pl-5">{mission.mission_name}</td>
          </tr>,
        );
      });
    }
    return myMissionsArr;
  };

  return (
    <div className="flex mx-20">
      <div className="missions container">
        <h2 className="font-bold text-3xl pb-3">My Missions</h2>
        <table className="border w-11/12">
          <tbody>
            <MyMissions missions={missions} />
          </tbody>
        </table>
      </div>
      <div className="rockets container">
        <h2 className="font-bold text-3xl pb-3">My Rockets</h2>
        <table className="border w-11/12">
          <tbody>
            {/* the next <tr> element should be replaced with the correct rockets component */}
            <tr className="border h-14">
              <td className="pl-5">TEST</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
}

export default MyProfile;
