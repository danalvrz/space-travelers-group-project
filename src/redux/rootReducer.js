import { combineReducers } from 'redux';
import missions from './missions/missions';
import rockets from './rockets/rockets';

const rootReducer = combineReducers({
  missions,
  rockets,
});

export default rootReducer;
