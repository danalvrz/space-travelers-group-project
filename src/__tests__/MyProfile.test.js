import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import MyProfile from '../components/MyProfile';

const mockStore = [];
const mockMissionsReducer = () => [{
  mission_id: 0, mission_name: 'test', mission_description: 'test', member: true,
}];
const mockRocketsReducer = () => mockStore;
const mockReducer = () => combineReducers({
  mockMissionsReducer,
  mockRocketsReducer,
});
const store = createStore(mockReducer);

describe('MyProfile section', () => {
  it('renders MyProfile correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <MyProfile />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
