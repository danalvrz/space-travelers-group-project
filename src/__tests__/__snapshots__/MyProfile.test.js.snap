// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile section renders MyProfile correctly 1`] = `
<div
  className="flex mx-20"
>
  <div
    className="missions container"
  >
    <h2
      className="font-bold text-3xl pb-3"
    >
      My Missions
    </h2>
    <table
      className="border w-11/12"
    >
      <tbody />
    </table>
  </div>
  <div
    className="rockets container"
  >
    <h2
      className="font-bold text-3xl pb-3"
    >
      My Rockets
    </h2>
    <table
      className="border w-11/12"
    >
      <tbody>
        <tr
          className="border h-14"
        >
          <td
            className="pl-5"
          >
            TEST
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
